### PR TITLE
Adding support for running a fully specified pipeline with no problem…

### PR DESCRIPTION
**Pull request by odin243**
… description specified

### CODE REVIEW

**Comment by odin243** [comment]
@cdbethune Can you take a look at this? One of the Simple-TA3 tests runs a pipeline with just a template, and no problem. This change appears to fix the reference errors in our TA2 code, however, the pipeline still does not run, as it appears the pipeline itself isn't passing the runtime's pipeline validation. That seems weird, and I'm still investigating it.

File: processing/pipeline.py [hunk 1]
```diff
@@ -53,6 +53,10 @@ def create(dataset_doc_path: str, problem: dict, prepend: pipeline.Pipeline=None
     # Load dataset in the same way the d3m runtime will
     train_dataset = dataset.Dataset.load(dataset_doc_path)
 
+    # If the pipeline is fully specified, just return it
```
Comment (cdbethune) [inline processing/pipeline.py]: I think there may be a misunderstanding of the meaning of `fully specified` vs. `prepend`.  A `prepend` is a series of primitives that a TA3 has requested be included at the start of the pipeline.  The TA2 is meant to append the pipeline it creates to those primitives.  A `fully specified` pipeline is one that is completely specified by the TA3, and requires nothing from the TA2 in terms of search - the TA2 is mean to execute it only.  The way a TA2 discriminates between the two cases is to look at the pipeline provided by the TA3 and check to see if it has a `PlaceholderStep` as its last stage.  If present, the TA2 will replace the `PlaceholderStep` with the pipeline it finds.  If there is no `PlaceholderStep` the TA2 should immediately return a search result - see https://gitlab.com/datadrivendiscovery/ta3ta2-api#fully-specified-pipelines.

The check being done below is not testing for the presence of a fully specified pipeline, but for the presence of a prepend sequence.  This probably isn't what you want if I understand the test case you're trying to address.

Comment (odin243) [reply processing/pipeline.py]: Probably poorly named, but in this case the prepend variable is the pipeline that was specified in the template field of the request, which appears to contain both pipelines with placeholders, as well as pipelines that are fully specified. As far as I can tell though, if there is a placeholder specified it is not valid to leave out the problem field - that combination appears to only be a valid API call if the TA2 is being asked to simply execute the primitives specified in the template without doing any searching.

Comment (cdbethune) [reply processing/pipeline.py]: I went back through the logic end to end, and I see what its doing now.  I think the check for the presence of the placeholder should still be made at the start of the `pipeline.create` function, as there is no point in doing any of the subsequent processing if the placeholder is present.  I've tested your changes against the TA3, and made a few additional tweaks to get the results back to the UI, including removing some redundant code in the `produce` task.  I'll add those changes directly to this branch.  If it all looks good to you, I think we can merge.

**Pull request merged**